Sources/Plasma/Apps/plClient/Mac-Cocoa/NSString+StringTheory.mm

```diff
@@ -40,13 +40,14 @@
 
 *==LICENSE==*/
 
-#include "StringTheory_NSString.h"
+#include "NSString+StringTheory.h"
+#include "hsDarwin.h"
 
 @implementation NSString (StringTheory)
 
 - (id)initWithSTString:(const ST::string&)string
 {
-    self = [self initWithUTF8String:string.c_str()];
+    self = NSStringCreateWithSTString(string);
```

Comment:
We're assuming ARC here, right? In non-ARC code, this would have to `release` the old `self`.

Reply:
All Obj-C code right now is ARC - so this should assume ARC.

release isn't typically called here because init is only allowed to be called as a constructor. It also might release the alloc'd memory. But I don't think I've ever considered it strongly because init is a constructor and should only be called once.

Reply:
would `self = [self initWithString:NSStringCreateWithSTString(string)];` be better? It feels like it's potentially duplicating the string, but maybe it's smart enough not to do that?

Reply:
Obj-C convention would be that the string should be duplicated - unless the function specifically calls that out in it's name. (EG the noCopy family of functions.)

I'd prefer duplication unless there's serious concerns about cost as it's usually the Obj-C default.

Reply:
Oh, I think I see what you're saying a bit clearer. Yeah, there might be two duplications. It's not super efficient. The initializing from the cstr would be preferable, but not super essential.

Reply:
Actually, is there a compelling reason to have the initializer function, and not just return `NSStringCreateWithSTString` from the static `stringWithSTString:` method?

Reply:
Self has already been allocated and has a pointer. Returning a different pointer will leak self I believe.

Reply:
I'm proposing this:

```diff
-- (id)initWithSTString:(const ST::string&)string
-{
-    self = NSStringCreateWithSTString(string);
-    return self;
-}
-
 + (id)stringWithSTString:(const ST::string&)string
 {
-    return [[NSString alloc] initWithSTString:string];
+    return NSStringCreateWithSTString(string);
 }
```

Reply:
Ah, I missed you said static. I think that would be ok.

Reply:
> Self has already been allocated and has a pointer. Returning a different pointer will leak self I believe.

That was my original worry - without ARC, the original `self` would need to be `release`d, otherwise it would leak. With ARC, everything should be fine. In any case, @dpogue's suggestion would be more readable, yeah. (`initWithSTString:` should never be necessary if we have `stringWithSTString:`.)